src/Compilers/CSharp/Portable/Binder/LockBinder.cs

```diff
@@ -64,13 +64,6 @@ internal override BoundStatement BindLockStatementParts(BindingDiagnosticBag dia
                 _ = diagnostics.ReportUseSite(lockTypeInfo.EnterScopeMethod, exprSyntax) ||
                     diagnostics.ReportUseSite(lockTypeInfo.ScopeType, exprSyntax) ||
                     diagnostics.ReportUseSite(lockTypeInfo.ScopeDisposeMethod, exprSyntax);
-
-                CheckRestrictedTypeInAsyncMethod(
```

Comment:
>CheckRestrictedTypeInAsyncMethod [](http://example.com/codeflow?start=16&length=32)

Are we skipping this check because `IDS_FeatureRefUnsafeInIteratorAsync` is added in the same release as `IDS_FeatureLockObject`?

Reply:
Yes.